pkg/plugins/manager/loader/finder/local_test.go

```diff
@@ -245,11 +247,46 @@ func TestFinder_Find(t *testing.T) {
 				},
 			},
 		},
+		{
+			name:        "Plugin with dist folder (core class)",
+			pluginDirs:  []string{filepath.Join(testData, "plugin-with-dist")},
```

Comment:
Just an FYI that I found a bug I introduced with https://github.com/grafana/grafana/pull/78042 where if you change this to `filepath.Join(testData, "plugin-with-dist", "plugin"), this will not prefer the `dist` version. The bug only happens when:
- Installing via the catalog (because of what path it specifies to load)
- The plugin has a `./dist/plugin.json` and at least one other plugin.json in its directory

Currently thinking on a potential fix